File: backend/schemas/export.py
"""
文件名：export.py
功能描述：导出相关数据模型。定义导出请求和响应格式。
作者：Claude Code
创建时间：2026-04-04
后续开发：Trea
TODO：
- [x] 核心数据模型已完成
"""

from enum import Enum
from pydantic import BaseModel, Field


class ExportFormat(str, Enum):
    """支持的导出格式"""
    WORD = "docx"
    PDF = "pdf"
    MARKDOWN = "md"
    HTML = "html"


class ExportRequest(BaseModel):
    """导出请求"""
    task_id: str = Field(description="任务 ID")
    format: ExportFormat = Field(description="导出格式")


class ExportResponse(BaseModel):
    """导出响应"""
    export_id: str = Field(description="导出任务 ID，用于下载")
    format: ExportFormat
    filename: str = Field(description="导出文件名")

File: frontend/src/api/export.ts
/**
 * 文件名：api/export.ts
 * 功能描述：文档导出 API 调用封装。
 * 作者：Trea
 * 创建时间：2026-04-04
 */

import client from './client'

export type ExportFormat = 'md' | 'docx' | 'html' | 'pdf'

export interface ExportRequest {
  task_id: string
  format: ExportFormat
}

export interface ExportResponse {
  export_id: string
  format: ExportFormat
  filename: string
}

export async function exportDocument(
  taskId: string,
  format: ExportFormat
): Promise<ExportResponse> {
  const response = await client.post('/export/', { task_id: taskId, format })
  return response.data
}

export async function downloadExport(exportId: string): Promise<Blob> {
  const response = await client.get(`/export/${exportId}/download`, {
    responseType: 'blob',
  })
  return response.data
}

export function downloadFile(blob: Blob, filename: string) {
  const url = window.URL.createObjectURL(blob)
  const link = document.createElement('a')
  link.href = url
  link.download = filename
  document.body.appendChild(link)
  link.click()
  document.body.removeChild(link)
  window.URL.revokeObjectURL(url)
}
